### PR TITLE
Fixed an issue with the contextualisation of XPaths with multiple predicates

### PR DESCRIPTION
**Pull request by rousso**
This was not coded correctly so it was getting wrong some interesting contextualisations. Have a look. I think it is pretty self explanatory.

### CODE REVIEW

File: src/test/java/eu/europa/ted/eforms/xpath/XPathStepTest.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ void testComparison_AddPredicates() {
 
     assertTrue(a.isTheSameAs(b));
 
-    assertTrue(a.isSimilarTo(b));
+    assertTrue(a.isSameAsOrNarrowerThan(b));
```
Comment (bertrand-lorentz) [inline src/test/java/eu/europa/ted/eforms/xpath/XPathStepTest.java]: With the new method name, this test seems incorrect. It says:
"/foo" is the same or narrower that "/foo[x=1][y=2]"

Or I don't understand what "is the same" means.

File: src/main/java/eu/europa/ted/eforms/xpath/XPathProcessor.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ private static String getContextualizedXpath(Queue<XPathStep> contextQueue,
       // the context is more restrictive than the path. In this case we want to use a dot step
       // with the predicate of the path.
```
Comment (bertrand-lorentz) [inline src/main/java/eu/europa/ted/eforms/xpath/XPathProcessor.java]: This comment is now confusing, as it says "the context is more restrictive than the path", but the "if" condition below checks that the path is "same or narrower" (= more restrictive ?) than the context.

**Review by bertrand-lorentz — COMMENTED** [review]
I am not sure this is correct, because the logic of XPathStep being "the same" is not clear to me.

File: src/test/java/eu/europa/ted/eforms/xpath/XPathStepTest.java (hunk 1 above)
Comment (rousso) [reply src/test/java/eu/europa/ted/eforms/xpath/XPathStepTest.java]: Indeed, you are right. I fixed it and requested another review.

File: src/main/java/eu/europa/ted/eforms/xpath/XPathProcessor.java (hunk 2 above)
Comment (rousso) [reply src/main/java/eu/europa/ted/eforms/xpath/XPathProcessor.java]: Fixed that too.

**Review by bertrand-lorentz — APPROVED** [review]

**Pull request merged**